fedimint-api-client/src/api/mod.rs

```diff
@@ -1046,28 +1003,235 @@ impl IClientConnection for Arc<WsClient> {
     }
 }
 
-pub type DynClientConnector = Arc<dyn IClientConnector>;
+#[allow(clippy::struct_excessive_bools)] // Shut up, Clippy
+pub struct EndpointRegistryBuilder {
```

Comment:
TODO: Add some docstring

Since `EndpointRegistry` is now built and injected into Clients, it gets a nice shiny builder and some docs.